### PR TITLE
drivers: platform: Switched to platform agnostic drivers

### PR DESCRIPTION
**Pull request by scuciurean**
Added the following platform agnostic drivers:
- spi xilinx & intel
- gpio xilinx & intel
- i2c intel

Updated readme and makefiles to use [driver]_extra.h types and remove platform_drivers.h

### CODE REVIEW

**Comment by scuciurean** [comment]
Changelog -> v2:

1) Removed unused labels in gpio.c
2) Added status check for gpios in ad9371
3) Fixed style issues

**Comment by scuciurean** [comment]
Changelog -> v3:

1. Fixed codacy label check

File: drivers/adc/ad7193/ad7193.h [hunk 1]
```diff
@@ -194,8 +194,8 @@ struct ad7193_init_param {
 	/* SPI */
 	spi_init_param	spi_init;
 	/* GPIO */
-	int8_t		gpio_cs;
-	int8_t		gpio_miso;
+	structgpio_init_param	gpio_cs;
+	structgpio_init_param	gpio_miso;
```
Comment (dbogdan) [inline drivers/adc/ad7193/ad7193.h]: structgpio_init_param - this is probably a typo.

**Comment by scuciurean** [comment]
Changelog -> v4:

1. Fixed struct typo in drivers/adc/ad7193/ad7193.h

**Comment by dbogdan** [comment]
Can you please fix the conflicts?

**Comment by amiclaus** [comment]
V4:

- resolve conflicts

**Pull request merged**